diff --git a/dlls/winegstreamer/media-converter/lib.c b/dlls/winegstreamer/media-converter/lib.c
index 4f7884c1178..bdbda22f697 100644
--- a/dlls/winegstreamer/media-converter/lib.c
+++ b/dlls/winegstreamer/media-converter/lib.c
@@ -306,30 +306,54 @@ void dump_fozdb_close(struct dump_fozdb *db)
 bool media_converter_init(void)
 {
     GST_DEBUG_CATEGORY_INIT(media_converter_debug,
-            "protonmediaconverter", GST_DEBUG_FG_YELLOW, "Proton media converter");
+                            "protonmediaconverter", GST_DEBUG_FG_YELLOW, "Proton media converter");

-    if (!GST_ELEMENT_REGISTER(protonvideoconverter, NULL))
+    // Check if the PROTON_VIDEO_CONVERT environment variable is set to 0
+    // Only try to register if the envvar isn't set
+    const char *proton_video_convert = getenv("PROTON_VIDEO_CONVERT");
+    if (!proton_video_convert || strcmp(proton_video_convert, "0") != 0)
     {
-        GST_ERROR("Failed to register protonvideoconverter.");
-        return false;
+        if (!GST_ELEMENT_REGISTER(protonvideoconverter, NULL))
+        {
+            GST_ERROR("Failed to register protonvideoconverter.");
+            return false;
+        }
     }

-    if (!GST_ELEMENT_REGISTER(protonaudioconverter, NULL))
+    // Check if the PROTON_AUDIO_CONVERT environment variable is set to 0
+    // Only try to register if the envvar isn't set
+    const char *proton_audio_convert = getenv("PROTON_AUDIO_CONVERT");
+    if (!proton_audio_convert || strcmp(proton_audio_convert, "0") != 0)
     {
-        GST_ERROR("Failed to register protonaudioconverter.");
-        return false;
+        if (!GST_ELEMENT_REGISTER(protonaudioconverter, NULL))
+        {
+            GST_ERROR("Failed to register protonaudioconverter.");
+            return false;
+        }
     }

-    if (!GST_ELEMENT_REGISTER(protonaudioconverterbin, NULL))
+    // Check if the PROTON_AUDIO_CONVERT_BIN environment variable is set to 0
+    // Only try to register if the envvar isn't set
+    const char *proton_audio_convert_bin = getenv("PROTON_AUDIO_CONVERT_BIN");
+    if (!proton_audio_convert_bin || strcmp(proton_audio_convert_bin, "0") != 0)
     {
-        GST_ERROR("Failed to register protonaudioconverterbin.");
-        return false;
+        if (!GST_ELEMENT_REGISTER(protonaudioconverterbin, NULL))
+        {
+            GST_ERROR("Failed to register protonaudioconverterbin.");
+            return false;
+        }
     }

-    if (!GST_ELEMENT_REGISTER(protondemuxer, NULL))
+    // Check if the PROTON_DEMUX environment variable is set to 0
+    // Only try to register if the envvar isn't set
+    const char *proton_demuxer = getenv("PROTON_DEMUX");
+    if (!proton_demuxer || strcmp(proton_demuxer, "0") != 0)
     {
-        GST_ERROR("Failed to register protondemuxer.");
-        return false;
+        if (!GST_ELEMENT_REGISTER(protondemuxer, NULL))
+        {
+            GST_ERROR("Failed to register protondemuxer.");
+            return false;
+        }
     }

     return true;
