--- a/sandbox/linux/system_headers/linux_seccomp.h
+++ b/sandbox/linux/system_headers/linux_seccomp.h
@@ -214,7 +214,9 @@
 #endif

-#ifndef SYS_SECCOMP
-#define SYS_SECCOMP                   1
-#endif
+#if !defined(SYS_SECCOMP) && !defined(__GLIBC_PREREQ)
+#define SYS_SECCOMP 1
+#elif !defined(SYS_SECCOMP) && defined(__GLIBC_PREREQ)
+#if !__GLIBC_PREREQ(2, 43)
+#define SYS_SECCOMP 1
+#endif
+#endif

 #endif  // SANDBOX_LINUX_SYSTEM_HEADERS_LINUX_SECCOMP_H_
