From 065d65617343175b93f09637e305b57df2d34370 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Tue, 11 Mar 2025 22:16:33 +0200
Subject: [PATCH] proton: fix logic error that caused proton to hang on x11

---
 proton | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/proton b/proton
index a8617299..145423db 100755
--- a/proton
+++ b/proton
@@ -1780,13 +1780,13 @@ if __name__ == "__main__":
     elif sys.argv[1] == "waitforexitandrun":
         #wait for wineserver to shut down
         g_session.run_proc([g_proton.wineserver_bin, "-w"])
-        winedrv = "x11"
-        if os.environ.get("PROTON_ENABLE_WAYLAND", None) == "1":
-            if os.environ.get("DISPLAY", None) and os.environ.get("WAYLAND_DISPLAY", None):
+        display_drv = "x11"
+        if os.environ.get("PROTON_ENABLE_WAYLAND", None) == "1" and os.environ.get("WAYLAND_DISPLAY", False):
+            display_drv = "x11,wayland"
+            if os.environ.get("DISPLAY", False):
                 del os.environ["DISPLAY"]
-            winedrv = "wayland"
         g_session.run_proc(
-            [g_proton.wine_bin, "reg.exe", "add", "HKCU\\Software\\Wine\\Drivers", "/v", "Graphics", "/d", f"{winedrv}", "/f"]
+            [g_proton.wine_bin, "reg.exe", "add", "HKCU\\Software\\Wine\\Drivers", "/v", "Graphics", "/d", f"{display_drv}", "/f"]
         )
         g_session.run_proc([g_proton.wineserver_bin, "-w"])
         #then run
-- 
2.48.1

