From f1afdb09d8e9743b7e74f7a8347a0617d8f75905 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Fri, 4 Apr 2025 00:39:17 +0300
Subject: [PATCH] build: add policy version minimum for CMake

---
 make/rules-cmake.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/make/rules-cmake.mk b/make/rules-cmake.mk
index ab898810..b9f4aaaa 100644
--- a/make/rules-cmake.mk
+++ b/make/rules-cmake.mk
@@ -14,6 +14,7 @@ $$(OBJ)/.$(1)-configure$(3): $$($(2)_SRC)/CMakeLists.txt
 
 	cd "$$($(2)_OBJ$(3))" && env $$($(2)_ENV$(3)) \
 	cmake "$$($(2)_SRC)" \
+	    -DCMAKE_POLICY_VERSION_MINIMUM=3.5 \
 	    -DCMAKE_INSTALL_PREFIX="$$($(2)_DST$(3))" \
 	    -DCMAKE_INSTALL_LIBDIR="lib$(subst 32,,$(3))" \
 	    -DCMAKE_BUILD_TYPE=plain \
-- 
2.49.0

