From 14e7468e9b5851cb935a5af05c20b41b05d0ad23 Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Wed, 14 May 2025 23:43:54 +0300
Subject: [PATCH 2/2] build: set preferred statck size for vkd3d

---
 Makefile.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile.in b/Makefile.in
index 1a7fcb99..334cae3e 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -903,6 +903,9 @@ VKD3D_CFLAGS = -fno-lto
 VKD3D_LDFLAGS = -static-libgcc $(CROSSLDFLAGS)
 VKD3D_DEPENDS = vulkan-headers spirv-headers
 
+VKD3D_i386_CFLAGS = -O3 -mpreferred-stack-boundary=2
+VKD3D_x86_64_CFLAGS = -O3
+
 VKD3D_i386_LIBDIR = $(VKD3D_i386_DST)/lib
 VKD3D_x86_64_LIBDIR = $(VKD3D_x86_64_DST)/lib
 
-- 
2.49.0

