From 4afe7dcbd917df9eb8a898c1bd09be51a054ee98 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Sun, 21 Apr 2024 09:40:42 +0200
Subject: [PATCH] disable-fastsync

Signed-off-by: Peter Jung <admin@ptr1337.dev>
---
 proton | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/proton b/proton
index d2ce8ea3..d7985d65 100755
--- a/proton
+++ b/proton
@@ -1216,6 +1216,10 @@ class Session:
         if not "WINE_CPU_TOPOLOGY" in self.env and appid in default_cpu_limit:
             self.env["WINE_CPU_TOPOLOGY"] = str(default_cpu_limit[appid])
 
+        #turn off fastsync by default unless it is enabled via env
+        if not "winesync" in self.compat_config:
+            self.compat_config.add("nowinesync")
+
     def init_wine(self):
         if "HOST_LC_ALL" in self.env and len(self.env["HOST_LC_ALL"]) > 0:
             #steam sets LC_ALL=C to help some games, but Wine requires the real value
@@ -1346,6 +1350,7 @@ class Session:
         self.check_environment("PROTON_NO_D3D10", "nod3d10")
         self.check_environment("PROTON_NO_ESYNC", "noesync")
         self.check_environment("PROTON_NO_FSYNC", "nofsync")
+        self.check_environment("PROTON_USE_WINESYNC", "winesync")
         self.check_environment("PROTON_FORCE_LARGE_ADDRESS_AWARE", "forcelgadd")
         self.check_environment("PROTON_OLD_GL_STRING", "oldglstr")
         self.check_environment("PROTON_HIDE_NVIDIA_GPU", "hidenvgpu")
@@ -1372,6 +1377,12 @@ class Session:
         else:
             self.env["WINEFSYNC"] = "1"
 
+        if "winesync" in self.compat_config:
+            self.env.pop("WINE_DISABLE_FAST_SYNC", "")
+        else:
+            if "nowinesync" in self.compat_config:
+                self.env["WINE_DISABLE_FAST_SYNC"] = "1"
+
         if "oldglstr" in self.compat_config:
             #mesa override
             self.env["MESA_EXTENSION_MAX_YEAR"] = "2003"
-- 
2.44.0

