From 18e94e63870ebcf79ebb42b7035cd3cb626ec090 Mon Sep 17 00:00:00 2001
From: Nickolay Shmyrev <nshmyrev@gmail.com>
Date: Fri, 23 Aug 2024 16:17:27 +0200
Subject: [PATCH] Fix name

---
 src/include/fst/bi-table.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/include/fst/bi-table.h b/src/include/fst/bi-table.h
index 7c5be23e5..45270011a 100644
--- a/src/include/fst/bi-table.h
+++ b/src/include/fst/bi-table.h
@@ -327,7 +327,7 @@ class VectorHashBiTable {
   }
 
   VectorHashBiTable(const VectorHashBiTable<I, T, S, FP, H, HS> &table)
-      : selector_(table.s_),
+      : selector_(table.selector_),
         fp_(table.fp_),
         h_(table.h_),
         id2entry_(table.id2entry_),
