From 879f09d2ac799cca99b78de3442194ebbe29d24a Mon Sep 17 00:00:00 2001
From: Nickolay Shmyrev <nshmyrev@gmail.com>
Date: Fri, 23 Aug 2024 16:10:49 +0200
Subject: [PATCH] Proper assign to unique_ptr

---
 src/include/fst/fst.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/include/fst/fst.h b/src/include/fst/fst.h
index 80e394808..57cf1fd68 100644
--- a/src/include/fst/fst.h
+++ b/src/include/fst/fst.h
@@ -701,8 +701,8 @@ class FstImpl {
     properties_.store(impl.properties_.load(std::memory_order_relaxed),
                       std::memory_order_relaxed);
     type_ = impl.type_;
-    isymbols_ = impl.isymbols_ ? impl.isymbols_->Copy() : nullptr;
-    osymbols_ = impl.osymbols_ ? impl.osymbols_->Copy() : nullptr;
+    isymbols_.reset(impl.isymbols_ ? impl.isymbols_->Copy() : nullptr);
+    osymbols_.reset(impl.osymbols_ ? impl.osymbols_->Copy() : nullptr);
     return *this;
   }
 
