From 8936cb1bcf8f8450f72c9c857b79d0d9d5af4c10 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Thu, 9 Jun 2022 02:48:06 +0200
Subject: [PATCH] fix missing import libcalamares, fix typo

---
 src/modules/initcpiocfg/main.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/initcpiocfg/main.py b/src/modules/initcpiocfg/main.py
index 6f94594aa..96258f632 100644
--- a/src/modules/initcpiocfg/main.py
+++ b/src/modules/initcpiocfg/main.py
@@ -11,7 +11,7 @@
 #
 #   Calamares is Free Software: see the License-Identifier above.
 #
-
+import libcalamares
 from libcalamares.utils import debug, target_env_call
 import os
 from collections import OrderedDict
@@ -127,7 +127,7 @@ def write_mkinitcpio_lines(hooks, modules, files, root_mount_point):
                 line = 'HOOKS="{!s}"'.format(' '.join(hooks))
             elif line.startswith("MODULES"):
                 line = 'MODULES="{!s}"'.format(' '.join(modules))
-            elif lines.startswith("FILES"):
+            elif line.startswith("FILES"):
                 line = 'FILES="{!s}"'.format(' '.join(files))
             mkinitcpio_file.write(line + "\n")
 
-- 
2.36.1

