diff '--color=auto' -rupN launcher.orig/justfile launcher/justfile
--- launcher.orig/justfile	2024-08-11 20:45:52.665174088 +0200
+++ launcher/justfile	2024-08-11 20:49:59.637999712 +0200
@@ -70,7 +70,7 @@ install-plugins:
         dest={{plugin-dir}}${plugin}
         mkdir -p ${dest}
         install -Dm0644 plugins/src/${plugin}/*.ron ${dest}
-        ln -sf {{bin-path}} {{plugin-dir}}${plugin}/$(echo ${plugin} | sed 's/_/-/')
+        ln -sf '/usr/bin/pop-launcher' {{plugin-dir}}${plugin}/$(echo ${plugin} | sed 's/_/-/')
     done
 
 # Install pop-launcher scripts
