diff --git a/bin_steam.sh b/bin_steam.sh
index fb81d47..134f94e 100755
--- a/bin_steam.sh
+++ b/bin_steam.sh
@@ -229,4 +229,20 @@
 # go to the install directory and run the client
 cd "$LAUNCHSTEAMDIR"
 
+STEAMDEV="$LAUNCHSTEAMDIR"/steam_dev.cfg
+
+if [[ ! -f "$STEAMDEV" ]]; then
+  touch "$STEAMDEV"
+fi
+
+if [[ -f "$STEAMDEV" ]]; then
+  if [[ -z $(cat "$STEAMDEV" | grep "nClientDownloadEnableHTTP2PlatformLinux") ]]; then
+       echo "@nClientDownloadEnableHTTP2PlatformLinux 0" >> "$STEAMDEV"
+  fi
+
+  if [[ -z $(cat "$STEAMDEV" | grep "fDownloadRateImprovementToAddAnotherConnection") ]]; then
+       echo "@fDownloadRateImprovementToAddAnotherConnection 1.0" >> "$STEAMDEV"
+  fi
+fi
+
 exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@"
