From 25973e884728233a86eade92a74ac4c347d8b48f Mon Sep 17 00:00:00 2001
From: Vasiliy Stelmachenok <ventureo@yandex.ru>
Date: Fri, 5 Jan 2024 18:44:25 +0300
Subject: [PATCH] Make it possible to build with LTO via Clang

---
 make/hotspot/lib/JvmFeatures.gmk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/make/hotspot/lib/JvmFeatures.gmk b/make/hotspot/lib/JvmFeatures.gmk
index 8eb57d765..f5eeddf48 100644
--- a/make/hotspot/lib/JvmFeatures.gmk
+++ b/make/hotspot/lib/JvmFeatures.gmk
@@ -175,6 +175,9 @@ ifeq ($(call check-jvm-feature, link-time-opt), true)
   else ifeq ($(call isCompiler, microsoft), true)
     JVM_CFLAGS_FEATURES += -GL
     JVM_LDFLAGS_FEATURES += -LTCG:INCREMENTAL
+  else ifeq ($(call isCompiler, clang), true)
+    JVM_CFLAGS_FEATURES += -flto=auto
+    JVM_LDFLAGS_FEATURES += $(CXX_O_FLAG_HIGHEST_JVM) -flto=auto
   endif
 else
   ifeq ($(call isCompiler, gcc), true)
-- 
2.43.0

