diff --git a/src/kosmickrisp/kosmicomp.c b/src/kosmickrisp/kosmicomp.c
index 501ba4e0095..a23aaff8dd7 100644
--- a/src/kosmickrisp/kosmicomp.c
+++ b/src/kosmickrisp/kosmicomp.c
@@ -45,7 +45,7 @@ static void
 debug_callback(void *priv, enum nir_spirv_debug_level debuglevel, size_t offset,
                const char *message)
 {
-   fprintf(stderr, "<%d> at %ld %s\n", debuglevel, offset, message);
+   fprintf(stderr, "<%d> at %zu %s\n", debuglevel, offset, message);
 }
 
 static int
