diff --git a/CMakeLists.txt b/CMakeLists.txt
index 85cd50d..558618c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@ project(PLASMALOGIN VERSION ${PROJECT_VERSION})
 option(INSTALL_PAM_CONFIGURATION "Install PAM configuration files" ON)

 # Version settings
-set(PROJECT_DEP_VERSION "6.5.90")
+set(PROJECT_DEP_VERSION "6.5.5")

 set(QT_MIN_VERSION "6.10.0")
 set(KF6_MIN_VERSION "6.22.0")
