From b01bfe6dbd4b9d0f07d30e53b1432bbf505b8426 Mon Sep 17 00:00:00 2001
From: Vicki Pfau <vi@endrift.com>
Date: Fri, 16 May 2025 18:28:42 -0700
Subject: [PATCH] Makefile: Remove making gamescope-session.service.wants
 directory

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f96420019a7f..fdbaae642fe9 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ install: target/release/steamos-manager target/release/steamosctl
 	install -d -m0755 "$(DESTDIR)/usr/share/dbus-1/system-services/"
 	install -d -m0755 "$(DESTDIR)/usr/share/dbus-1/system.d/"
 	install -d -m0755 "$(DESTDIR)/usr/lib/systemd/system/"
-	install -d -m0755 "$(DESTDIR)/usr/lib/systemd/user/gamescope-session.service.wants/"
+	install -d -m0755 "$(DESTDIR)/usr/lib/systemd/user/"
 
 	install -Ds -m755 "target/release/steamos-manager" "$(DESTDIR)/usr/lib/steamos-manager"
 	install -D -m755 "target/release/steamosctl" "$(DESTDIR)/usr/bin/steamosctl"
-- 
2.49.0

