From 3328a33510ec2b079527bc911600997bc4e3e30b Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Wed, 16 Apr 2025 08:06:48 +0800
Subject: [PATCH] Add support for gamescope-session-steam

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 data/user/steamos-manager.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/user/steamos-manager.service b/data/user/steamos-manager.service
index c147e403693b..27623cc7eaf3 100644
--- a/data/user/steamos-manager.service
+++ b/data/user/steamos-manager.service
@@ -1,6 +1,6 @@
 [Unit]
 Description=SteamOS Manager Daemon
-After=gamescope.service
+After=gamescope-session-plus@steam.service
 
 [Service]
 Type=dbus
-- 
2.49.0

