From a464a0034348d23cf69e406a82497244c6ea1037 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Wed, 16 Apr 2025 08:06:48 +0800
Subject: [PATCH 1/2] Add support for gamescope-session-steam

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 data/system/steamos-manager.service | 2 --
 1 file changed, 2 deletions(-)

diff --git a/data/system/steamos-manager.service b/data/system/steamos-manager.service
index e49ea157d6a3..f44c5725e1eb 100644
--- a/data/system/steamos-manager.service
+++ b/data/system/steamos-manager.service
@@ -1,7 +1,5 @@
 [Unit]
 Description=SteamOS Manager Daemon
-Wants=steamos-log-submitter.service
-After=steamos-log-submitter.service
 StartLimitIntervalSec=120
 StartLimitBurst=5
 
-- 
2.51.2


From d380aee2fdfb7a53f78ddf3607feb788eef1b00c Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Fri, 17 Oct 2025 14:33:15 +0800
Subject: [PATCH 2/2] Makefile: Do not install orca.service

orca 49.0+ comes with its own service file and conflicts with the one
from steamos-manager. To fix the conflict, remove the one from
steamos-manager and use the service from the orca package.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1a33cf26f282..60e51f1a8b69 100644
--- a/Makefile
+++ b/Makefile
@@ -43,4 +43,3 @@ install: target/release/steamos-manager target/release/steamosctl
 	install -m644 "data/user/com.steampowered.SteamOSManager1.service" "$(DESTDIR)/usr/share/dbus-1/services/"
 	install -m644 "data/user/steamos-manager.service" "$(DESTDIR)/usr/lib/systemd/user/"
 	install -m644 "data/user/steamos-manager-session-cleanup.service" "$(DESTDIR)/usr/lib/systemd/user/"
-	install -m644 "data/user/orca.service" "$(DESTDIR)/usr/lib/systemd/user/"
-- 
2.51.2

