--- a/make/rules-cmake.mk	2023-10-12 16:17:36.260323696 +0200
+++ b/make/rules-cmake.mk	2023-10-12 16:17:49.534329154 +0200
@@ -16,7 +16,7 @@
 	cmake "$$($(2)_SRC)" \
 	    -DCMAKE_INSTALL_PREFIX="$$($(2)_DST$(3))" \
 	    -DCMAKE_INSTALL_LIBDIR="lib$(subst 32,,$(3))" \
-	    -DCMAKE_BUILD_TYPE=plain \
+	    -DCMAKE_BUILD_TYPE=release \
 	    -DCMAKE_SYSTEM_NAME=$(if $(4),Windows,) \
 	    -DCMAKE_SHARED_LIBRARY_PREFIX_C=$(if $(4),,lib) \
 	    -DCMAKE_IMPORT_LIBRARY_PREFIX_C=$(if $(4),,lib) \


--- a/make/rules-meson.mk	2023-10-11 15:59:05.000000000 +0200
+++ b/make/rules-meson.mk	2023-10-12 16:27:43.666517375 +0200
@@ -50,7 +50,7 @@
 	meson "$$($(2)_OBJ$(3))" "$$($(2)_SRC)" \
 	      --prefix="$$($(2)_DST$(3))" \
 	      --libdir="lib$(subst 32,,$(3))" \
-	      --buildtype=plain \
+	      --buildtype=release \
 	      $(if $(4),--cross-file="$$($(2)_OBJ$(3))/cross-$(3).txt",) \
 	      $$($(2)_MESON_ARGS) \
 	      $$($(2)_MESON_ARGS$(3)) \
