From 3656401958f409a9a012d19ed1347c7cc819e5ae Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Fri, 25 Sep 2020 18:49:22 +0200
Subject: [PATCH] src: Add QPainterPath

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 src/qroundprogressbar.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/qroundprogressbar.cpp b/src/qroundprogressbar.cpp
index ea1bf93..a056bf3 100755
--- a/src/qroundprogressbar.cpp
+++ b/src/qroundprogressbar.cpp
@@ -20,6 +20,7 @@
 #include "qroundprogressbar.h"
 
 #include <QtGui/QPainter>
+#include <QtGui/QPainterPath>
 
 
 QRoundProgressBar::QRoundProgressBar(QWidget *parent) :
-- 
2.28.0.585.ge1cfff6765

