From bf89ad1a3f322df8803fcbaecf664699aac5244e Mon Sep 17 00:00:00 2001
From: Tobias Powalowski <tpowa@archlinux.org>
Date: Fri, 11 Mar 2022 15:47:59 +0100
Subject: [PATCH] add real symlink support

---
 functions | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/functions b/functions
index 48cfd7a..5584a80 100644
--- a/functions
+++ b/functions
@@ -571,7 +571,15 @@ add_file() {
     else
         quiet "adding file: %s" "$dest"
     fi
-    command install -Dm$mode "$src" "$BUILDROOT$dest"
+   #check if $src is a symlink
+    if [[ -L $src ]]; then
+        # add the symlink
+        add_file "$(realpath -- "$src")"
+        # create the symlink
+        add_symlink "$src" "$(realpath -- "$src")"
+    else
+       command install -Dm$mode "$src" "$BUILDROOT$dest"
+    fi
 }
 
 add_runscript() {
@@ -631,7 +639,15 @@ add_binary() {
     mode=$(stat -c %a "$binary")
 
     # always add the binary itself
-    add_file "$binary" "$dest" "$mode"
+    # check if binary is a symlink
+    if [[ -L $binary ]]; then
+        # add the binary
+        add_file "$(realpath -- "$binary")"
+        # create the symlink
+        add_symlink "$binary" "$(realpath -- "$binary")"
+    else
+        add_file "$binary" "$dest" "$mode"
+    fi
 
     # negate this so that the RETURN trap is not fired on non-binaries
     ! lddout=$(ldd "$binary" 2>/dev/null) && return 0
