diff --git a/mkinitcpio b/mkinitcpio
index 0f4a90ae9678..fde108b4983e 100755
--- a/mkinitcpio
+++ b/mkinitcpio
@@ -777,6 +777,7 @@ preload_builtin_modules() {
 
     if [[ -r $_d_kmoduledir/modules.builtin.modinfo ]]; then
         while IFS=.= read -rd '' modname field value; do
+            [[ -z "$modname" ]] && continue
             _addedmodules[${modname//-/_}]=2
             case "$field" in
                 alias)
