diff --git a/mkinitcpio.d/example.preset b/mkinitcpio.d/example.preset
index db97f3f..22b1ddf 100644
--- a/mkinitcpio.d/example.preset
+++ b/mkinitcpio.d/example.preset
@@ -1,7 +1,7 @@
 # Example mkinitcpio preset file

 # preset names
-PRESETS=('default' 'fallback')
+PRESETS=('default')

 # ALL_kver is used if presetname_kver is not set
 # The kver can be specified as an explicit kernel version or
diff --git a/mkinitcpio.d/hook.preset b/mkinitcpio.d/hook.preset
index 1bb9aa4..195edbc 100644
--- a/mkinitcpio.d/hook.preset
+++ b/mkinitcpio.d/hook.preset
@@ -3,7 +3,7 @@
 #ALL_config="/etc/mkinitcpio.conf"
 ALL_kver="/boot/vmlinuz-%PKGBASE%"

-PRESETS=('default' 'fallback')
+PRESETS=('default')

 #default_config="/etc/mkinitcpio.conf"
 default_image="/boot/initramfs-%PKGBASE%.img"
