diff --git a/subprojects/wlroots/types/output/render.c b/subprojects/wlroots/types/output/render.c
index a185d844..59faabd8 100644
--- a/subprojects/wlroots/types/output/render.c
+++ b/subprojects/wlroots/types/output/render.c
@@ -1,3 +1,4 @@
+#include <unistd.h>
 #include <assert.h>
 #include <drm_fourcc.h>
 #include <stdlib.h>
@@ -13,6 +14,18 @@
 #include "render/pixel_format.h"
 #include "types/wlr_output.h"
 
+static bool is_nvidia_card = false;
+static bool nv_initialized = false;
+
+static bool get_nvidia_check(void) {
+	if (!nv_initialized) {
+		is_nvidia_card = (access("/sys/module/nvidia", F_OK) != -1);
+		nv_initialized = true;
+	}
+
+	return is_nvidia_card;
+}
+
 bool wlr_output_init_render(struct wlr_output *output,
 		struct wlr_allocator *allocator, struct wlr_renderer *renderer) {
 	assert(allocator != NULL && renderer != NULL);
@@ -240,6 +253,10 @@ bool output_pick_format(struct wlr_output *output,
 }
 
 uint32_t wlr_output_preferred_read_format(struct wlr_output *output) {
+	if (get_nvidia_check()) {
+		return DRM_FORMAT_XRGB8888;
+	}
+
 	struct wlr_renderer *renderer = output->renderer;
 	assert(renderer != NULL);
 
