diff --git a/src/modules/partition/gui/ChoicePage.cpp b/src/modules/partition/gui/ChoicePage.cpp
index f98ce4df6..daf9fd352 100644
--- a/src/modules/partition/gui/ChoicePage.cpp
+++ b/src/modules/partition/gui/ChoicePage.cpp
@@ -298,7 +298,7 @@ ChoicePage::setupChoices()
     }
 
     m_itemsLayout->addWidget( m_alongsideButton );
-    m_itemsLayout->addWidget( m_replaceButton );
+    //m_itemsLayout->addWidget( m_replaceButton );
     m_itemsLayout->addWidget( m_eraseButton );
 
     m_somethingElseButton = new PrettyRadioButton;
@@ -1252,6 +1252,7 @@ describePartitionTypes( const QStringList& types )
 void
 ChoicePage::setupActions()
 {
+    m_replaceButton->hide();
     Logger::Once o;
 
     Device* currentDevice = selectedDevice();
@@ -1305,11 +1306,11 @@ ChoicePage::setupActions()
             cDebug() << Logger::SubEntry << "contains resizable" << it;
             atLeastOneCanBeResized = true;
         }
-        if ( PartUtils::canBeReplaced( *it, o ) )
+        /*if ( PartUtils::canBeReplaced( *it, o ) )
         {
             cDebug() << Logger::SubEntry << "contains replaceable" << it;
             atLeastOneCanBeReplaced = true;
-        }
+        }*/
         if ( ( *it )->isMounted() )
         {
             atLeastOneIsMounted = true;
@@ -1430,7 +1431,7 @@ ChoicePage::setupActions()
 
     if ( atLeastOneCanBeReplaced )
     {
-        m_replaceButton->show();
+        //m_replaceButton->show();
     }
     else
     {
