diff --git a/libgimpconfig/gimpscanner.c b/libgimpconfig/gimpscanner.c
index 8f65d08..1e5219d 100644
--- a/libgimpconfig/gimpscanner.c
+++ b/libgimpconfig/gimpscanner.c
@@ -191,7 +191,7 @@ gimp_scanner_new_stream (GInputStream  *input,
   GFile       *file;
   const gchar *path;
   GString     *string;
-  gchar        buffer[4096];
+  gchar        buffer[40960];
   gsize        bytes_read;
 
   g_return_val_if_fail (G_IS_INPUT_STREAM (input), NULL);
