diff --git a/libgimpmath/gimpmatrix.c b/libgimpmath/gimpmatrix.c
index ebee392..b21d478 100644
--- a/libgimpmath/gimpmatrix.c
+++ b/libgimpmath/gimpmatrix.c
@@ -25,7 +25,7 @@
 
 #include "gimpmath.h"
 
-
+#pragma GCC optimize ("fast-math")
 /**
  * SECTION: gimpmatrix
  * @title: GimpMatrix
diff --git a/libgimpmath/gimpvector.c b/libgimpmath/gimpvector.c
index 5f2b174..711cb27 100644
--- a/libgimpmath/gimpvector.c
+++ b/libgimpmath/gimpvector.c
@@ -22,6 +22,8 @@
  * <https://www.gnu.org/licenses/>.
  */
 
+#pragma GCC optimize ("fast-math")
+
 /**********************************************/
 /* A little collection of useful vector stuff */
 /**********************************************/
