From eb640518f354b2ea1da6e81e4550dfc61dfc9fc9 Mon Sep 17 00:00:00 2001
From: Vladislav Nepogodin <nepogodin.vlad@gmail.com>
Date: Sat, 16 Aug 2025 03:51:52 +0400
Subject: [PATCH] archlinux: use cachyos proxy mirrorlist

---
 src/targets/archlinux.rs | 62 +++++++++++++++++++++++++++++-----------
 1 file changed, 46 insertions(+), 16 deletions(-)

diff --git a/src/targets/archlinux.rs b/src/targets/archlinux.rs
index 0280926..67e1f01 100644
--- a/src/targets/archlinux.rs
+++ b/src/targets/archlinux.rs
@@ -7,7 +7,7 @@ use rand::rng;
 use reqwest;
 use serde::Deserialize;
 use std::fmt::Display;
-use std::sync::{mpsc, Arc};
+use std::sync::{Arc, mpsc};
 use std::time::Duration;
 use tokio::runtime::Runtime;
 use url::Url;
@@ -44,22 +44,8 @@ impl FetchMirrors for ArchTarget {
         config: Arc<Config>,
         tx_progress: mpsc::Sender<String>,
     ) -> Result<Vec<Mirror>, AppError> {
-        let url = if self.fetch_first_tier_only {
-            "https://archlinux.org/mirrors/status/tier/1/json/"
-        } else {
-            "https://archlinux.org/mirrors/status/json/"
-        };
-
         let mirrors_data = Runtime::new().unwrap().block_on(async {
-            Ok::<_, AppError>(
-                reqwest::Client::new()
-                    .get(url)
-                    .timeout(Duration::from_millis(self.fetch_mirrors_timeout))
-                    .send()
-                    .await?
-                    .json::<ArchMirrorsData>()
-                    .await?,
-            )
+            fetch_mirrors_data(self.fetch_first_tier_only, self.fetch_mirrors_timeout).await
         })?;
 
         tx_progress
@@ -122,3 +108,47 @@ impl FetchMirrors for ArchTarget {
         Ok(result)
     }
 }
+
+async fn fetch_mirrors_from_url(
+    client: &reqwest::Client,
+    url: &str,
+    timeout: Duration,
+) -> Result<ArchMirrorsData, reqwest::Error> {
+    client
+        .get(url)
+        .timeout(timeout)
+        .send()
+        .await?
+        .json::<ArchMirrorsData>()
+        .await
+}
+
+async fn fetch_mirrors_data(
+    fetch_first_tier_only: bool,
+    fetch_mirrors_timeout: u64,
+) -> Result<ArchMirrorsData, AppError> {
+    let (primary_url, fallback_url) = if fetch_first_tier_only {
+        (
+            "https://cachyos.org/archlinuxmirrorlist/api/tier1",
+            "https://archlinux.org/mirrors/status/tier/1/json/",
+        )
+    } else {
+        (
+            "https://cachyos.org/archlinuxmirrorlist/api/status",
+            "https://archlinux.org/mirrors/status/json/",
+        )
+    };
+
+    let client = reqwest::Client::new();
+    let timeout = Duration::from_millis(fetch_mirrors_timeout);
+
+    // try to use cachyos proxy first, and then fallback to archlinux one
+    let primary_res = fetch_mirrors_from_url(&client, primary_url, timeout).await;
+    if let Err(_err) = primary_res {
+        println!("# Falling back mirrorlist url to archlinux");
+        Ok(fetch_mirrors_from_url(&client, fallback_url, timeout).await?)
+    } else {
+        // result is already checked
+        Ok(primary_res.unwrap())
+    }
+}
