diff --git a/IGC/ElfPackager/main.cpp b/IGC/ElfPackager/main.cpp
index 5a421895d..2139fec59 100644
--- a/IGC/ElfPackager/main.cpp
+++ b/IGC/ElfPackager/main.cpp
@@ -107,8 +107,7 @@ std::unique_ptr<IGCLLVM::Module> LocalCloneModule(
         if (VMap.find(&*I) == VMap.end())
         {
             GlobalVariable *GV = new GlobalVariable(*New,
-                IGCLLVM::getNonOpaquePtrEltTy(I->getType()),
-                I->isConstant(), I->getLinkage(),
+                I->getValueType(), I->isConstant(), I->getLinkage(),
                 (Constant*) nullptr, I->getName(),
                 (GlobalVariable*) nullptr,
                 I->getThreadLocalMode(),
