From bcc24cd744ea64fa07bb04271303b155bddbf161 Mon Sep 17 00:00:00 2001
From: Vasiliy Stelmachenok <ventureo@cachyos.org>
Date: Thu, 9 Jan 2025 14:13:16 +0300
Subject: [PATCH] scripts/chwd: Wrong order of arguments [Hotfix]

Signed-off-by: Vasiliy Stelmachenok <ventureo@cachyos.org>
---
 scripts/chwd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/chwd b/scripts/chwd
index 5f03634..9f0bd9b 100755
--- a/scripts/chwd
+++ b/scripts/chwd
@@ -122,7 +122,7 @@ local function remove(packages)
     end
 end
 
-local function exec_hook(name, hooks)
+local function exec_hook(hooks, name)
     local hook = hooks[name]
 
     if not hook then
