diff --git a/profiles/pci/handhelds/profiles.toml b/profiles/pci/handhelds/profiles.toml
index c6828769e824..9c89ab383d61 100644
--- a/profiles/pci/handhelds/profiles.toml
+++ b/profiles/pci/handhelds/profiles.toml
@@ -75,32 +75,7 @@ vendor_ids = "1002"
 device_ids = "15bf"
 hwd_product_name_pattern = '(83E1)'
 priority = 6
-packages = 'hhd hhd-ui adjustor cachyos-handheld mesa lib32-mesa vulkan-radeon lib32-vulkan-radeon opencl-mesa lib32-opencl-mesa'
-post_install = """
-    echo "Legion go chwd installing..."
-    username=$(id -nu 1000)
-    services=("hhd@${username}")
-    echo "Enabling services..."
-    for service in ${services[@]}; do
-        systemctl enable --now "${service}.service"
-    done
-
-    if [ -d /etc/limine-entry-tool.d ]; then
-        cat << 'EOF' > /etc/limine-entry-tool.d/20-chwd.conf
-KERNEL_CMDLINE[default]+="amdgpu.sg_display=0"
-EOF
-    fi
-"""
-post_remove = """
-    echo "Legion go chwd removing..."
-    username=$(id -nu 1000)
-    services=("hhd@${username}")
-    echo "Disabling services..."
-    for service in ${services[@]}; do
-        systemctl disable "${service}.service"
-    done
-    rm -f /etc/limine-entry-tool.d/20-chwd.conf
-"""
+packages = 'inputplumber steamos-manager steamos-powerbuttond cachyos-handheld mesa lib32-mesa vulkan-radeon lib32-vulkan-radeon opencl-mesa lib32-opencl-mesa'
 
 [amd-legion-go-s]
 desc = 'Lenovo Legion Go S'
