diff --git a/build/moz.configure/lto-pgo.configure b/build/moz.configure/lto-pgo.configure
index 2f32b11588..db3202d9b8 100644
--- a/build/moz.configure/lto-pgo.configure
+++ b/build/moz.configure/lto-pgo.configure
@@ -259,9 +259,9 @@ def lto(value, c_compiler, ld64_known_good, target, instrumented_build):
             # recommends this as the "generic 64-bit specific x86 processor model":
             #
             # https://github.com/llvm/llvm-project/blob/e7694f34ab6a12b8bb480cbfcb396d0a64fe965f/llvm/lib/Target/X86/X86.td#L1165-L1187
             if target.cpu == "x86_64":
-                ldflags.append("-mllvm:-mcpu=x86-64")
+                ldflags.append("-mllvm:-mcpu=x86-64-v3")
             # We do not need special flags for arm64.  Hooray for fixed-length
             # instruction sets.
         else:
             num_cores = multiprocessing.cpu_count()
