diff --git a/xxhash.h b/xxhash.h
index dceb921..1446b15 100644
--- a/xxhash.h
+++ b/xxhash.h
@@ -5067,6 +5067,29 @@ XXH3_scrambleAcc_avx512(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
     }
 }
 
+__attribute__((target("avx512f,avx512dq")))
+XXH_FORCE_INLINE XXH_TARGET_AVX512 void
+XXH3_scrambleAcc_avx512dq(void* XXH_RESTRICT acc, const void* XXH_RESTRICT secret)
+{
+    XXH_ASSERT((((size_t)acc) & 63) == 0);
+    XXH_STATIC_ASSERT(XXH_STRIPE_LEN == sizeof(__m512i));
+    {   __m512i* const xacc = (__m512i*) acc;
+        
+        const __m512i prime64 = _mm512_set1_epi64((unsigned long long)XXH_PRIME32_1);
+
+        /* xacc[0] ^= (xacc[0] >> 47) */
+        __m512i const acc_vec     = *xacc;
+        __m512i const shifted     = _mm512_srli_epi64    (acc_vec, 47);
+        
+        /* xacc[0] ^= secret; */
+        __m512i const key_vec     = _mm512_loadu_si512   (secret);
+        __m512i const data_key    = _mm512_ternarylogic_epi32(key_vec, acc_vec, shifted, 0x96);
+
+        /* xacc[0] *= XXH_PRIME32_1; */
+        *xacc = _mm512_mullo_epi64(data_key, prime64);
+    }
+}
+
 XXH_FORCE_INLINE XXH_TARGET_AVX512 void
 XXH3_initCustomSecret_avx512(void* XXH_RESTRICT customSecret, xxh_u64 seed64)
 {
@@ -6158,7 +6181,11 @@ typedef void (*XXH3_f_initCustomSecret)(void* XXH_RESTRICT, xxh_u64);
 
 #define XXH3_accumulate_512 XXH3_accumulate_512_avx512
 #define XXH3_accumulate     XXH3_accumulate_avx512
-#define XXH3_scrambleAcc    XXH3_scrambleAcc_avx512
+#if defined(__AVX512DQ__)
+    #define XXH3_scrambleAcc    XXH3_scrambleAcc_avx512dq
+#else
+    #define XXH3_scrambleAcc    XXH3_scrambleAcc_avx512
+#endif
 #define XXH3_initCustomSecret XXH3_initCustomSecret_avx512
 
 #elif (XXH_VECTOR == XXH_AVX2)
