From 726f989199169aad8a26620ef9f5a575e52d13c7 Mon Sep 17 00:00:00 2001
From: Vladislav Nepogodin <nepogodin.vlad@gmail.com>
Date: Sun, 2 Jul 2023 19:30:02 +0400
Subject: [PATCH] =?utf-8?q?=F0=9F=9A=A7=20fix=20compilation?=
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index f9bb7df..2574185 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -135,7 +135,7 @@ $(TUNER_LIBS): $(OPATH)libbpftune.so $(BPF_SKELS) $(TUNER_OBJS)
 	$(CC) $(CFLAGS) -shared -o $(@) $(patsubst $(OPATH)%.so,%.c,$(@)) \
 		$(LDLIBS) -lbpftune $(LDFLAGS)
 
-$(OPATH)libbpftune.o: probe.skel.h probe.skel.legacy.h
+$(OPATH)libbpftune.o: probe.skel.h $(BPF_SKELS)
 	$(QUIET_CC)$(CC) $(CFLAGS) -c libbpftune.c -o $@
 
 $(OPATH)bpftune.o: $(OPATH)libbpftune.so
-- 
2.41.0

