diff --git a/sample_tuner/Makefile b/sample_tuner/Makefile
index 6161de3..82be56d 100644
--- a/sample_tuner/Makefile
+++ b/sample_tuner/Makefile
@@ -37,7 +37,7 @@ installprefix = $(DESTDIR)/$(prefix)
 
 INSTALLPATH = $(installprefix)
 
-CFLAGS = -fPIC -Wall -Wextra -march=native -g -I../include -std=c99
+CFLAGS = -fPIC -Wall -Wextra -march=x86-64 -g -I../include -std=c99
 
 CFLAGS += -DBPFTUNE_VERSION='"$(BPFTUNE_VERSION)"' $(INCLUDES)
 
diff --git a/src/Makefile b/src/Makefile
index bb662e3..2ed2b09 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -55,7 +55,7 @@ endif
 VERSION = 0.1.3
 VERSION_SCRIPT  := libbpftune.map
 
-CFLAGS = -fPIC -Wall -Wextra -march=native -g -I../include -std=c99
+CFLAGS = -fPIC -Wall -Wextra -march=x86-64 -g -I../include -std=c99
 
 CFLAGS += -DBPFTUNE_VERSION='"$(BPFTUNE_VERSION)"' $(INCLUDES)
 
diff --git a/test/strategy/Makefile b/test/strategy/Makefile
index 45150eb..9b40ba4 100644
--- a/test/strategy/Makefile
+++ b/test/strategy/Makefile
@@ -38,7 +38,7 @@ installprefix = $(DESTDIR)/$(prefix)
 
 INSTALLPATH = $(installprefix)
 
-CFLAGS = -fPIC -Wall -Wextra -march=native -g -I../include -std=c99
+CFLAGS = -fPIC -Wall -Wextra -march=x86-64 -g -I../include -std=c99
 
 CFLAGS += -DBPFTUNE_VERSION='"$(BPFTUNE_VERSION)"' $(INCLUDES)
