From c36d6d913e805107134a74bbca7aa185c3903669 Mon Sep 17 00:00:00 2001
From: Vasiliy Stelmachenok <ventureo@cachyos.org>
Date: Wed, 28 May 2025 16:15:08 +0300
Subject: [PATCH] Fix compilation with GCC 15

Signed-off-by: Vasiliy Stelmachenok <ventureo@cachyos.org>
---
 kernel/Kbuild | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/Kbuild b/kernel/Kbuild
index 84a9d43..eecc1a0 100644
--- a/kernel/Kbuild
+++ b/kernel/Kbuild
@@ -66,6 +66,7 @@ ccflags-y += -D__KERNEL__ -DMODULE -DNVRM -DNV_VERSION_STRING=\"470.256.02\" -Wn
 ccflags-y += $(call cc-option,-Werror=undef,)
 ccflags-y += -DNV_SPECTRE_V2=$(NV_SPECTRE_V2)
 ccflags-y += -DNV_KERNEL_INTERFACE_LAYER
+ccflags-y += -std=gnu17
 
 #
 # Detect SGI UV systems and apply system-specific optimizations.
-- 
2.49.0

